src/Package/Impl/Shell/VsMainThread.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Windows.Threading;
+using Microsoft.Common.Core.Threading;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.R.Package.Shell {
+    internal sealed class VsMainThread : IMainThread {
+        private readonly Dispatcher _mainThreadDispatcher;
+        private readonly Thread _mainThread;
+
+        public VsMainThread() {
+            _mainThread = Thread.CurrentThread;
```

Comment:
While we are at the very beginning, let's try to use `_mainThread = ThreadHelper.JoinableTaskContext.MainThread;`